voxblox/include/voxblox/core/layer_inl.h

```diff
@@ -49,6 +50,29 @@ void Layer<VoxelType>::getProto(LayerProto* proto) const {
 }
 
 template <typename VoxelType>
+Layer<VoxelType>::Layer(const Layer& other) {
+  voxel_size_ = other.voxel_size_;
+  voxels_per_side_ = other.voxels_per_side_;
+  block_size_ = other.block_size_;
+  block_size_inv_ = other.block_size_inv_;
+
+  for (const typename BlockHashMap::value_type& key_value_pair :
+       other.block_map_) {
+    const BlockIndex& block_idx = key_value_pair.first;
+    const typename BlockType::Ptr& block_ptr = key_value_pair.second;
+
+    typename BlockType::Ptr new_block = allocateBlockPtrByIndex(block_idx);
+
+    for (size_t linear_idx = 0u; linear_idx < block_ptr->num_voxels();
```

Comment:
Not important, but I wonder if it's better to have a cloneBlock or copyBlockData or something function within block, where we just copy the complete corresponding memory over. Otherwise 2 function calls and voxel-by-voxel copy seems like a lot of overhead if this is used commonly.